pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -71,42 +73,72 @@ func (p *MellanoxPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetwork
 	needReboot = false
 	err = nil
 	attributesToChange = map[string]mlnxNic{}
+	mellanoxNicsStatus = map[string]map[string]*sriovnetworkv1.InterfaceExt{}
+	mellanoxNicsSpec = map[string]*sriovnetworkv1.Interface{}
+	processedNics := map[string]bool{}
 
+	for _, iface := range new.Status.Interfaces {
+		if iface.Vendor != MellanoxVendorId {
+			continue
+		}
+
+		pciPrefix := getPciAddressPrefix(iface.PciAddress)
+		if ifaces, ok := mellanoxNicsStatus[pciPrefix]; ok {
+			ifaces[iface.PciAddress] = &iface
+		} else {
+			mellanoxNicsStatus[pciPrefix] = map[string]*sriovnetworkv1.InterfaceExt{iface.PciAddress: &iface}
+		}
+	}
+
+	// Add only mellanox cards that required changes in the map, to help track dual port NICs
```

Comment:
In case of an interface config is removed in the new spec, what shall the plugin do for both single port and dual port NICs?

Reply:
I am working on another PR that handle that will do regular check when the `totalVFs!= numVfs ` (single), `totalVfs != max(pf0Vfs,pf1Vfs)` then change the totalVfs.

Mellanox plugin only check for the new Specs

Reply:
@pliurh The operator changes the NumVfs for devices that doesn't have a policy to be 0 always, right?
If true then I don't need to check for the `otherIfaceStatus.NumVfs`, if there is no spec it will be 0

Reply:
Right. Users shall always do that by removing the policy. But as you don't reset the NUM_OF_VF in the firmware, this value can only increase. It would work, but is that what you want?

Reply:
yes, if there is no spec then there is no need to have VFs and the total vfs should reset to 0